client/a8c-for-agencies/sections/partner-directory/agency-expertise/hooks/use-expertese-form-validation.ts

```diff
@@ -0,0 +1,82 @@
+import { useTranslate } from 'i18n-calypso';
+import { useCallback, useState } from 'react';
+import { AgencyDirectoryApplication } from '../../types';
+import { areURLsUnique } from '../../utils/tools';
+
+export const CAPTURE_URL_RGX =
+	/^(http:\/\/www\.|https:\/\/www\.|http:\/\/|https:\/\/)?[a-z0-9]+([-.][a-z0-9]+)*\.[a-z]{2,63}(:[0-9]{1,5})?(\/.*)?$/i;
+
+const isValidUrl = ( urlString: string ) => {
+	if ( ! urlString || urlString === '' ) {
+		return false;
+	}
+	return !! CAPTURE_URL_RGX.test( urlString );
+};
+
+type ValidationState = {
+	services?: string;
+	products?: string;
+	directories?: string;
+	clientSites?: string;
+	feedbackUrl?: string;
+};
+
+const useExperteseFormValidation = () => {
```

Comment:
`useExpertiseFormValidation` and the file name: `use-expertise-form-validation.ts`